### PR TITLE
chore: Migrated from long-term IAM to OIDC

### PR DESCRIPTION
**Pull request by kg-aws**
*Issue #, if available:*

*Description of changes:*
Migrating GitHub Action repository from using the long-term AWS IAM credentials to OIDC.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test-workflow.yml [hunk 1]
```diff
@@ -20,9 +24,9 @@ jobs:
 
     - name: Configure AWS credentials
       uses: aws-actions/configure-aws-credentials@v4
-      with:
-        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
-        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      with:
```
Comment (omkhegde) [inline test-workflow.yml:27]: nit: please remove the extra spacing at the end of this line.

Comment (kg-aws) [reply test-workflow.yml]: Addressed the feedback.

**Review by omkhegde — APPROVED** [review]

**Review by amazreech — COMMENTED** [review]
I believe these changes should also be reflected on the `.github/workflows/test-workflow.yml` file?

**Comment by kg-aws** [comment]
> I believe these changes should also be reflected on the `.github/workflows/test-workflow.yml` file?

I thought the same initially but the [script](https://github.com/aws-actions/amazon-ecs-render-task-definition/blob/integ-tests/run_integ_test.sh#L10) copies the file to `.github/workflows/test-workflow.yml`

**Comment by amazreech** [comment]
> I thought the same initially but the [script](https://github.com/aws-actions/amazon-ecs-render-task-definition/blob/integ-tests/run_integ_test.sh#L10) copies the file to `.github/workflows/test-workflow.yml`

This is true, but I was thinking for consistency sake we should update these at the same time. The updating of the other file would happen only when a commit is merged on the master branch, which might not happen soon, and could cause confusion/inconsistency? 

Do let me know what you think or you feel we could wait for the next commit to be merged and this file to be updated.

**Review by amazreech — APPROVED** [review]

**Comment by kg-aws** [comment]
> > I thought the same initially but the [script](https://github.com/aws-actions/amazon-ecs-render-task-definition/blob/integ-tests/run_integ_test.sh#L10) copies the file to `.github/workflows/test-workflow.yml`
> 
> This is true, but I was thinking for consistency sake we should update these at the same time. The updating of the other file would happen only when a commit is merged on the master branch, which might not happen soon, and could cause confusion/inconsistency?
> 
> Do let me know what you think or you feel we could wait for the next commit to be merged and this file to be updated. Or maybe a commit (prolly existing dependabot PRs) to update this?

I think it's a good point about confusion in case not commit goes into the `master` branch for a while. I updated the PR to include the changes in the `.github/workflows` folder.

**Pull request merged**